controllers/model/storage/storage_provider.go

```diff
@@ -0,0 +1,28 @@
+package storage
+
+import (
+	modelv1alpha1 "github.com/alibaba/kubedl/apis/model/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+)
+
+var StorageProviders = make(map[string]StorageProvider)
+
+func init() {
+	StorageProviders["AliCloudNas"] = NewAliCloudNasProvider()
+	StorageProviders["LocalStorage"] = NewLocalStorageProvider()
+
+}
+
+type StorageProvider interface {
+	CreatePersistentVolume(model *modelv1alpha1.Storage, pvName string) *v1.PersistentVolume
+}
+
+func GetStorageProvider(storage *modelv1alpha1.Storage) StorageProvider {
```

Comment:
Does AliCloudNas have a higher priority than LocalStorage? If each model/modelVersion only uses one of the two choices, shall we add an attribute, say StorageType within the structure Storage in the crd definition?

Reply:
we can have a validation check to check only one storage type is specified. 
Like in PersistentVolumeSource, we don't need a explicit volume type